File: ITG Hiring Challenge Front-end/src/services/BrandService.js
import axiosInstance from "@/config/axiosInstance";

class BrandService {
  getByCategoryId(categoryId) {
    return axiosInstance.get(
      `/brands/findByCategoryId?categoryId=${categoryId}`
    );
  }
}

const brandService = new BrandService();

export default brandService;

File: src/main/java/com/biricik/automotive/business/abstracts/BrandService.java
package com.biricik.automotive.business.abstracts;

import java.util.List;

import com.biricik.automotive.business.requests.brandRequest.CreateBrandRquest;
import com.biricik.automotive.business.responses.brandResponses.CreateBrandResponse;
import com.biricik.automotive.business.responses.brandResponses.FindBrandByCategoryIdResponse;
import com.biricik.automotive.model.Brand;

public interface BrandService {

	
	
	List<FindBrandByCategoryIdResponse> findByCategoryId(int categoryID);

	Brand findById(int brandId);

	CreateBrandResponse addBrand(CreateBrandRquest createBrandRquest);
	
}
